sdk/storage/storage-blob/src/BlobClient.ts

```diff
@@ -1101,4 +1104,30 @@ export class BlobClient extends StorageClient {
     }
     await batch.do();
   }
+
+  /**
+   * ONLY AVAILABLE IN NODE.JS RUNTIME.
+   *
+   * Downloads an Azure Blob to a local file.
+   * Fails if the the given file path already exits.
+   * Offset and count are optional, pass 0 and undefined respectively to download the entire blob.
+   *
+   * @param {string} filePath
+   * @param {number} [offset] From which position of the block blob to download.
+   * @param {number} [count] How much data to be downloaded. Will download to the end when passing undefined.
+   * @param {BlobDownloadOptions} [options] Options to Blob download options.
+   * @returns {Promise<void>}
+   * @memberof BlobClient
+   */
+  public async downloadToFile(
+    filePath: string,
+    offset: number = 0,
+    count?: number,
+    options?: BlobDownloadOptions
+  ): Promise<void> {
```

Comment:
Return return the original `download` response

Reply:
I am following the pattern of the high level functions.  For example, `downloadToBuffer` also returns `Promise<void>`.

Reply:
It's not same, all original highlevel functions are for high performance concurrency upload download.  `downloadToBuffer` is implemented as concurrency download which makes it unavailable to return a response from GetBlob requests. 

`downloadToFile` is a simpler wrapper of `download`, which is a single request. (when no retry happens).

Reply:
Fair.  One more question: are we okay with returning the response, with the readableStreamBody possibly not resettable? i.e., impossible to rewind and re-read the stream?

Reply:
Good question, `readableStreamBody` is not usable. Can we override it to `undefined` before returning to customers?

Reply:
Good idea! Done. Had to set it's underlying stream since `readableStreamBody` is a getter.